src/controllers/bucket-controller.js

```diff
@@ -0,0 +1,68 @@
+const { BucketModel, ProductModel } = require('../models');
+const { NotFoundError, ValidationError } = require('../errors');
+
+const { ResponseHandlerUtil } = require('../utils');
+
+async function addProductToBucket(req, res, next) {
+  const { productId, quantity } = req.body;
+  const userId = req.userData._id;
+
+  try {
+    const product = await ProductModel.findOne({ _id: productId });
+    const increaseAmount = product.price * quantity;
+
+    let bucketData;
+
+    if (!product) {
```

Comment:
check before getting some value from it, otherwise you'll get cannot read price from null error